packages/suite/src/middlewares/wallet/coinjoinMiddleware.ts

```diff
@@ -69,5 +69,15 @@ export const coinjoinMiddleware =
             }
         }
 
+        if (action.type === SUITE.TOR_STATUS) {
+            if (['Disabling', 'Disabled', 'Error'].includes(action.payload)) {
+                api.dispatch(coinjoinAccountActions.pauseInterruptAllCoinjoinSessions());
+            }
+            // We restore sessions that were interrupted when successfully Enabled, not when Enabling.
+            if (action.payload === 'Enabled') {
```

Comment:
check race conditions with remembered device

Reply:
I checked for race conditions when reloading and did not happen.